Clarification on static token kubeconfig docs (#7856)
gardener/gardener: docs/usage/shoot_access.md

Shoots have to "opt-in" for such defaulting by using the `oidc=enable` label.

For further information on `(Cluster)OpenIDConnectPreset`, refer to [ClusterOpenIDConnectPreset and OpenIDConnectPreset](openidconnect-presets.md).

> ## Static Token kubeconfig

Reviewer: If you are marking this option as deprecated, then consider:
```suggestion
## Static Token kubeconfig (deprecated)
```

> > **Note:** Static token kubeconfig is deprekated for Shoot clusters using Kubernetes version >= 1.27

Reviewer: I don't think it is correct to say that this option is deprecated only for Shoots using K8s >= 1.27.
According to my understanding we consider the option as deprecated today no matter which K8s version is used. Simply, starting K8s >= 1.27 we lock the field to false - validations should ensure that the static token is disabled for K8s >= 1.27 Shoots. So both things are different - deprecation != enforced validation.

And when we mark something as deprecated, we have to say what is the alternative.

```suggestion
> **Note:** Static token kubeconfig is deprecated. The [`shoots/adminkubeconfig` subresource](#shootsadminkubeconfig-subresource) should be used instead.
```

I don't know if it makes sense to repeat the K8s 1.27 part as it is already written below.

Author: For shoots with Kubernetes < 1.27 the config in the shoot spec for the enableStaticTokenKubeconfig in still considered.

Reviewer: IMO, deprecation != default values && enforced validation. Default values && enforced validation are a way we phase out the static token kubeconfig option in a gradual manner.
It is wrong to say "static token kubeconfig is deprecated for Shoots clusters using K8s > 1.27" - it is simply forbidden to use static token kubeconfig with K8s >= 1.27. Deprecated means "it is not recommended to use it, there is a new/better alternative, but you should be able to still use it if you want but have in mind that it will be removed in the future".

> > **Note:** Static token kubeconfig is deprekated for Shoot clusters using Kubernetes version >= 1.27

Reviewer 2: ```suggestion
> **Note:** Static token kubeconfig is deprecated for Shoot clusters using Kubernetes version >= 1.27
```

> > **Note:** Static token kubeconfig is not available for Shoot clusters created with Kubernetes version >= 1.27. The [`shoots/adminkubeconfig` subresource](#shootsadminkubeconfig-subresource) should be used instead.

Reviewer: Btw, it is not only about Shoot clusters created with Kubernetes version >= 1.27. It should also forbid updating from K8s 1.26 to 1.27 when enableStaticTokenKubeconfig=true. Anyways, this validation is about to be added as part of https://github.com/gardener/gardener/issues/7783.

```suggestion
> **Note:** Static token kubeconfig is not available for Shoot clusters with Kubernetes version >= 1.27. The [`shoots/adminkubeconfig` subresource](#shootsadminkubeconfig-subresource) should be used instead.
```
or 

```suggestion
> **Note:** Static token kubeconfig is not available for Shoot clusters using Kubernetes version >= 1.27. The [`shoots/adminkubeconfig` subresource](#shootsadminkubeconfig-subresource) should be used instead.
```
